src/main/java/cloud/eppo/BaseEppoClient.java

```diff
@@ -66,9 +68,33 @@ protected BaseEppoClient(
       String host,
       AssignmentLogger assignmentLogger,
       BanditLogger banditLogger,
+      IConfigurationStore configurationStore,
+      boolean isGracefulMode,
+      boolean expectObfuscatedConfig) {
+    this(
+        apiKey,
+        sdkName,
+        sdkVersion,
+        host,
+        assignmentLogger,
+        banditLogger,
+        configurationStore,
+        isGracefulMode,
+        expectObfuscatedConfig,
+        null);
+  }
+
+  protected BaseEppoClient(
```

Comment:
given this constructor is protected and will only be used by upstream SDK builders (and tests here), I don't think we need to make `initialConfiguration` optional and can just require `null` or a `CompletableFuture` to be passed in to a single constructor.